### PR TITLE
Added player list API 

### PR DESCRIPTION
**Pull request by tig567899**
Adds the prerequisite for being able to view list of all players in the mobile application, requested by numerous alpha testers.

### CODE REVIEW

**Review by firemaster1294 — APPROVED** [review]
Looks good, but with one comment: how will this handle mod “scores”? Currently it doesn’t seem to process these which is fine, but idk if this is something we want

File: API Documentation.md
Comment (pkrshnan) [inline API Documentation.md]: Hm I agree with this API showing a -1 if the score is hidden, but in the app I think we should show a 0 for the score. The reason for this is that it might give extra information we don't want the players to have, if that makes sense.

File: app/serializers/player_serializer.py
Comment (pkrshnan) [inline app/serializers/player_serializer.py]: What does this line achieve and why do we need it? I may be missing something here.

**Review by pkrshnan — APPROVED** [review]
This LGTM as long as it's been tested. Just one concern on how we show hidden score in the UI.

File: API Documentation.md
Comment (tig567899) [reply API Documentation.md]: I mean, in the app, we can just write "Hidden". The purpose of returning -1 is that anyone who uses the API gets the relevant information. How we display that information is up to whoever is accessing the API. At the moment, we only have one type of client (the app), but if someone else decides to build something for this, it would be up to them.

File: app/serializers/player_serializer.py
Comment (tig567899) [reply app/serializers/player_serializer.py]: My bad. Removing it now.

**Comment by tig567899** [comment]
> Looks good, but with one comment: how will this handle mod “scores”? Currently it doesn’t seem to process these which is fine, but idk if this is something we want

That's actually a good point, I forgot to include the identities of moderators in the return data.

**Comment by tig567899** [comment]
Added in moderator and spectator data. The client side can either merge them into the same list like we do on the website, or leave them separate. Either way, it's their call.

**Pull request merged**
